### PR TITLE
cucumber-core: Use the docstring content type in the json formatter.

### PR DESCRIPTION
**Pull request by rjwittams**
## Summary

Docstring content type is currently not carried down into the json formatter. 
This worked in 1.2.5.
PickleString did not have the corresponding field, so I added that in:
https://github.com/cucumber/cucumber/pull/292
(*The mono repo PR is required for this one to compile, so this will fail CI*). 

https://github.com/cucumber/cucumber-jvm/issues/1263
Fixing this bug. 

Passes the contentType field down into the map that gets json serialized.
Uses the content_type name to match 1.2.5 and other language reports.

## Motivation and Context

https://github.com/cucumber/cucumber-jvm/issues/1263

## How Has This Been Tested?

Unit test added. 
All tests run, Pax Exam Calculator Test appears to be broken before this change, due to a deployment issue?

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [X] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

I do not know if any docs need changing. I expect not as this is a regression from 1.2.5

### CODE REVIEW

**Comment by mlvandijk** [comment]
@rjwittams Thanks for submitting a PR for this issue. Unfortunately the build fails due to a compilation error in cucumber-core: 
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /home/travis/build/cucumber/cucumber-jvm/core/src/main/java/cucumber/runtime/formatter/JSONFormatter.java:[253,51] cannot find symbol
symbol: method getContentType()
location: variable docString of type gherkin.pickles.PickleString

**Comment by rjwittams** [comment]
I'm not sure if this is an automatic reply. It fails because the cucumber mono repo needs to apply its PR before this will work, as stated above. I didn't want to mess about with the versions in pom.xml before it made sense.

File: core/src/test/java/cucumber/runtime/formatter/JSONFormatterTest.java [hunk 1]
```diff
@@ -678,6 +678,68 @@ public void should_format_scenario_with_a_step_with_a_doc_string() throws Throwa
     }
 
     @Test
+    public void should_format_scenario_with_a_step_with_a_doc_string2() throws Throwable {
```
Comment (mpkorstanje) [inline core/src/test/java/cucumber/runtime/formatter/JSONFormatterTest.java]: I'd suggest `should_format_scenario_with_a_step_with_a_doc_string_and_content_type`

**Comment by rjwittams** [comment]
Sorry, I did have it at that name before, but was fighting with the json pretty formatter, and recreated the method a few times. Will update.

**Comment by mlvandijk** [comment]
@rjwittams Not a bot ;) Sorry, I missed your comment about the PRs being related.

**Comment by brasmusson** [comment]
Should cucumber/cucumber#292 be accepted and the content type of doc strings be included in the Pickles, then it is convenient for the Json Formatter to get in from the Pickle, until then #1309 is fixing the regression from v1.x.x so that the content type is included in the Json Formatter output.

**Comment by brasmusson** [comment]
cucumber/cucumber#292 has been accepted, now we just have to wait for a Gherkin release before this PR can be merged.

**Comment by aslakhellesoy** [comment]
Hi @rjwittams,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**